install_gridpack.sh

```diff
@@ -1,103 +1,139 @@
 
-# Create directory for installing external packages
-if test -z ${GP_EXT_DEPS}
-then
-     export GP_EXT_DEPS=${GRIDPACK_ROOT_DIR}/external-dependencies
-fi
+  # remove existing build and install dir
+  rm -rf "$gridpack_build_dir"
+  rm -rf "$gridpack_install_dir"
 
-cd ${GRIDPACK_ROOT_DIR}
+  # create the build dir
+  mkdir -p "$gridpack_build_dir"
 
-# Set environment variable GRIDPACK_BUILD_DIR and create a build directory
-export GRIDPACK_BUILD_DIR=${GRIDPACK_ROOT_DIR}/src/build
+  pushd "$gridpack_build_dir" || exit
 
-GRIDPACK_INSTALL_DIR=${GRIDPACK_ROOT_DIR}/src/install
-export GRIDPACK_DIR=${GRIDPACK_INSTALL_DIR}
-
-if ${install_gridpack}
-then
-
-  rm -rf $GRIDPACK_BUILD_DIR
-  mkdir $GRIDPACK_BUILD_DIR
-
-  rm -rf ${GRIDPACK_INSTALL_DIR}
-
-  cd ${GRIDPACK_BUILD_DIR}
-    
-  ## GridPACK installation
-  echo "Building GridPACK"
-
-#  git checkout develop
+  # todo? git checkout develop
 
+  # remove existing cmake output
   rm -rf CMake*
 
-  cmake_args="-D GA_DIR:STRING=${GP_EXT_DEPS}/ga-5.8/install_for_gridpack \
-   -D BOOST_ROOT:STRING=${GP_EXT_DEPS}/boost_1_78_0/install_for_gridpack \    
-   -D Boost_DIR:STRING=${GP_EXT_DEPS}/boost_1_78_0/install_for_gridpack/lib/cmake/Boost-1.78.0 \
-   -D Boost_LIBRARIES:STRING=${GP_EXT_DEPS}/boost_1_78_0/install_for_gridpack/lib \   
-   -D Boost_INCLUDE_DIRS:STRING=${GP_EXT_DEPS}/boost_1_78_0/install_for_gridpack/include \
-   -D PETSC_DIR:PATH=${GP_EXT_DEPS}/petsc/install_for_gridpack \                  
-   -D MPI_CXX_COMPILER:STRING='mpicxx' \     
-   -D MPI_C_COMPILER:STRING='mpicc' \                                         
-   -D MPIEXEC:STRING='mpiexec' \                                                            
-   -D GRIDPACK_TEST_TIMEOUT:STRING=30 \                                                     
-   -D CMAKE_INSTALL_PREFIX:PATH=${GRIDPACK_INSTALL_DIR} \             
-   -D CMAKE_BUILD_TYPE:STRING=Debug \
-   -D BUILD_SHARED_LIBS=YES \
-   -D Boost_NO_SYSTEM_PATHS:BOOL=TRUE \
-    ..   "
- 
-  cmake ${cmake_args}                                                                                     
-
-   echo "Installing GridPACK develop branch"
-   make -j 10 install
-fi
-
-if ${install_gridpack_python}
-then
-    echo "Installing GridPACK python wrapper"
-    
-    cd ${GRIDPACK_ROOT_DIR}
-    
-    git submodule update --init
-
-    echo ${GRIDPACK_DIR}
-    cd python
-
-    export RHEL_OPENMPI_HACK=yes
-
-    rm -rf build
-    
-    ${python_exe} setup.py build
-
-    rm -rf ${GRIDPACK_INSTALL_DIR}/lib/python
-    mkdir ${GRIDPACK_INSTALL_DIR}/lib/python
-    
-    PYTHONPATH="${GRIDPACK_DIR}/lib/python:${PYTHONPATH}"
-    export PYTHONPATH
-    ${python_exe} setup.py install --home="$GRIDPACK_DIR"
-    
-fi
-
-cd ${GRIDPACK_ROOT_DIR}
+
+  # load module related to mpi4py
+  case $distribution in
+    fedora | rhel | centos | rocky)
+      echo "Loading mpi4py module"
+      module load "mpi/openmpi-$(arch)"
+      ;;
```

Comment:
Just WRT the echoed message: the `module` command just puts the MPI compiler wrappers on the $PATH.  mpi4py was handled when the `python3-mpi4py-openmpi` package was installed.  

This should have been done before Boost was built too.  

Note that the `module` command only affects the current shell.   If you only want do do it once, it really needs to be done in the top-most shell (`Dockerfile`).